jobs: TestRegistryGC failed

jobs.TestRegistryGC [failed](https://tanzanite.cluster.engflow.com/invocations/default/74e807d2-2fb6-494d-9dce-f748f9ec8fd4?testReportRun=3&testReportShard=3&testReportAttempt=1#targets-Ly9wa2cvam9iczpqb2JzX3Rlc3Q=) on master @ [435a330ddf15f70234747ef55d3d3721a09ba252](https://github.com/cockroachdb/cockroach/commits/435a330ddf15f70234747ef55d3d3721a09ba252):


```
=== RUN   TestRegistryGC
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestRegistryGC1850856216
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    registry_test.go:243: query 'SELECT id FROM system.jobs ORDER BY id': expected:
        103
        931290392835293185
        931290398844125185
        931290403497115649
        931290406873792513
        931290416589766657
        931290425002688513
        931290434696609793
        931290444903219201
        931290453484470273
        931290462941511681
        
        got:
        103
        931290392835293185
        931290398844125185
        931290403497115649
        931290406873792513
        931290410707386369
        931290416589766657
        931290425002688513
        931290425804161025
        931290434696609793
        931290435940417537
        931290444903219201
        931290445410664449
        931290453052948481
        931290453484470273
        931290462941511681
        
    panic.go:541: -- test log scope end --
test logs left over in: outputs.zip/logTestRegistryGC1850856216
--- FAIL: TestRegistryGC (80.31s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=3</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRegistryGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35099

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -205,7 +205,7 @@ INSERT INTO t."%s" VALUES('a', 'foo');
 
 		var id jobspb.JobID
 		db.QueryRow(t,
-			`INSERT INTO system.jobs (status, created) VALUES ($1, $2) RETURNING id`, status, created).Scan(&id)
+			`INSERT INTO system.jobs (status, created, job_type) VALUES ($1, $2, 'SCHEMA CHANGE') RETURNING id`, status, created).Scan(&id)
 		db.Exec(t, `INSERT INTO system.job_info (job_id, info_key, value) VALUES ($1, $2, $3)`, id, GetLegacyPayloadKey(), payload)
 		db.Exec(t, `INSERT INTO system.job_info (job_id, info_key, value) VALUES ($1, $2, $3)`, id, GetLegacyProgressKey(), progress)
 		return strconv.Itoa(int(id))
@@ -239,9 +239,9 @@ INSERT INTO t."%s" VALUES('a', 'foo');
 			newCanceledJob := writeJob("new_canceled", earlier, earlier.Add(time.Minute),
 				StatusCanceled, mutOptions)
 
-			sqlActivityJob := fmt.Sprintf("%d", SqlActivityUpdaterJobID)
-			db.CheckQueryResults(t, `SELECT id FROM system.jobs ORDER BY id`, [][]string{
-				{sqlActivityJob}, {oldRunningJob}, {oldSucceededJob}, {oldFailedJob}, {oldRevertFailedJob}, {oldCanceledJob},
+			selectJobsQuery := `SELECT id FROM system.jobs WHERE job_type = 'SCHEMA CHANGE' ORDER BY id`
+			db.CheckQueryResults(t, selectJobsQuery, [][]string{
+				{oldRunningJob}, {oldSucceededJob}, {oldFailedJob}, {oldRevertFailedJob}, {oldCanceledJob},
 				{newRunningJob}, {newSucceededJob}, {newFailedJob}, {newRevertFailedJob}, {newCanceledJob}})
 
 			testutils.SucceedsSoon(t, func() error {
@@ -251,19 +251,19 @@ INSERT INTO t."%s" VALUES('a', 'foo');
 				return nil
 			})
 
-			db.CheckQueryResults(t, `SELECT id FROM system.jobs ORDER BY id`, [][]string{
-				{sqlActivityJob}, {oldRunningJob}, {oldRevertFailedJob}, {newRunningJob},
+			db.CheckQueryResults(t, selectJobsQuery, [][]string{
+				{oldRunningJob}, {oldRevertFailedJob}, {newRunningJob},
 				{newSucceededJob}, {newFailedJob}, {newRevertFailedJob}, {newCanceledJob}})
 
 			if err := s.JobRegistry().(*Registry).cleanupOldJobs(ctx, ts.Add(time.Minute*-10)); err != nil {
 				t.Fatal(err)
 			}
-			db.CheckQueryResults(t, `SELECT id FROM system.jobs ORDER BY id`, [][]string{
-				{sqlActivityJob}, {oldRunningJob}, {oldRevertFailedJob}, {newRunningJob}, {newRevertFailedJob}})
+			db.CheckQueryResults(t, selectJobsQuery, [][]string{
+				{oldRunningJob}, {oldRevertFailedJob}, {newRunningJob}, {newRevertFailedJob}})
 
 			// Delete the revert failed, and running jobs for the next run of the
 			// test.
-			_, err := sqlDB.Exec(`DELETE FROM system.jobs WHERE id = $1 OR id = $2 OR id = $3 OR id = $4`,
+			_, err := sqlDB.Exec(`DELETE FROM system.jobs WHERE id IN ($1, $2, $3, $4)`,
 				oldRevertFailedJob, newRevertFailedJob, oldRunningJob, newRunningJob)
 			require.NoError(t, err)
 		}

